src/utils/SparseGrid.h

```diff
@@ -27,8 +27,10 @@ class SparseGridInvasive
      *
      * \param[in] cell_size The size to use for a cell (square) in the grid.
      *    Typical values would be around 0.5-2x of expected query radius.
+     * \param[in] elem_reserve Number of elements to research space for.
+     * \param[in] max_load_factor Maximum average load factor before rehashing.
      */
-    SparseGridInvasive(coord_t cell_size);
+    SparseGridInvasive(coord_t cell_size, size_t elem_reserve=0U, float max_load_factor=1.0f);
```

Comment:
Note: I included the max_load_factor because in the (horrible) master implementation of BucketGrid2D I needed a rather high load factor to prevent constant rehashing. I misused the bucket implementation of unordered_map and introduced hash collisions on purpose. I therefore had a high load factor on purpose. 
Now that each cell maps to one container of multiple items, rather than all those items themselves we don't have that problem any more.